src/Jaeger.Core/Senders/SenderResolver.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Jaeger.Senders
+{
+    /// <summary>
+    /// Provides a way to resolve an appropriate <see cref="ISender"/>.
+    /// </summary>
+    public class SenderResolver
+    {
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger _logger;
+        private readonly IDictionary<string, ISenderFactory> _senderFactories;
```

Comment:
Trying to understand this part. In Java the "resolver" pattern does not require explicit registration, but auto-discovers the factories on the classpath. Is the intention the same here, or is manual registration always required? In which case, why support more than one registered factory?

Reply:
As written in the `Things taken into account` section of the first post, auto-discovery sadly not works with ASP.NET Core since it's not loading assemblies not referenced by the code (in comparison to .NET Core and .NET Framework). Therefore it's always required to keep it the same over all implementations.

Java also supports multiple senders here: https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-core/src/main/java/io/jaegertracing/internal/senders/SenderResolver.java#L56

So I made sure to use the same JAEGER_SENDER_FACTORY environment variable. If only one is registered, it's not necessary to define it and just the one will be used. This should be the common use case. I liked the dictionary more than just an enumerable as mine is not generated by auto-discovery, as it helps with resolving the one set with JAEGER_SENDER_FACTORY.